File: Java/2do año/Leccion07/JavaBeans/src/domain/Persona.java

package domain;

import java.io.Serializable;

public class Persona implements Serializable{
    private String nombre;
    private String apellido;

    //Constructor vacio: esto es obligatorio
    public Persona() {
    
    }
    public Persona(String nombre, String apellido) {
    this.nombre = nombre;
    this.apellido = apellido;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    @Override
    public String toString() {
        return "Persona{" + "nombre=" + nombre + ", apellido=" + apellido + '}';
    }
    
    
}

File: JavaScript/Tecnicatura3js/TSLeccion07/Persona.js
class Persona{

    static contadorPersonas = 0;

    constructor(nombre, apellido, edad){
        this._idPersona = ++Persona.contadorPersonas
        this._nombre = nombre;
        this._apellido = apellido;
        this._edad = edad;
    }

    get idPersona(){
        return this._idPersona
    }

    get nombre(){
        this._nombre;
    }

    set nombre(nombre){
        this._nombre =  nombre;
    }

    get apellido(){
        return this._apellido
    }

    set apellido(apellido){
        this._apellido
    }

    get edad(){
        this._edad = edad
    }

    set edad(edad){
        this._edad = edad
    }

    toString (){
        return `
        ${this._idPersona} 
        ${this._nombre}
        ${this.apellido} 
        ${this._edad}`;
    }

    
}
